### ISSUE TITLE
AssertionError in RSA Public Key Encryption - Not sure how to handle

### ISSUE THREAD

**Issue by tulpn** [pyca/cryptography#2933]
Hi Guys, 

I have the following PEM formated Public Key with which I'd like to encrypt some string:

```
-----BEGIN PUBLIC KEY-----MIICIjANBgkqhkiG9w0BAQEFAAOCAg8AMIICCgKCAgEAy2SBwad@TZRyCPDUdxZkGqBipyY7UmTTdGOKL/dgOOsuITXFy/COCi2ylo/owI7u@/K5lkdoBunn1OEu@UFzxRy4bKQK2OkpP6dtAHfyexracpeiHVccRIkQB743DpfupDVtR5TweZsjkRM2SDcpwXZJbnSpa8pzxl6mHvZhV/D2kDkSPLQnI6iWchExt5e/psorpWvy22smaL1xvINztaaRSXFqMQILtSxv/2lJJGc6PRyQacQ4XxDk2DwKBKPyFYHYelYpjGFErUIRz7/EdY4vc3pzr2HMo0AFvvu51ezKIrFGDXt9I7xwEEOSY59mNtA9u9oPyd7UhXkK1JB7OSI458rfHZEXUsvmqqpHH8FKC4dBCZasEHqzMdter05h/v2hMbPV/QgqrArs1p/P6KMpnlYvhnTrIj9p30kOIq7WSfCyXBlTjAb8unYST6vtHXB21TOM6e6szea8fWgCnO3e/xD4M/xdVEqgGxDJWr7qOeHL@l1I93f0wBpO4qlmdMgFz@UqO2p@7dtN4JQO5PwCYHOLgE9bJJZKzaB61BeM4xznqiVfUxOBmtN0k5m1DnKL0lUagSyitwq0H7AY802O91n7ykFYO/gsFkfqNFtdgsGX56lqILU1dYKX3fVsd8r5rENndBE7J0HEUjYz9JHjS@m8EMJ@57kKztP0EgUCAwEAAQ==-----END PUBLIC KEY-----
```

I guess it is not in the correct format, so I had to adjust it on the fly. 

```py
pem_public_key = pem_public_key.replace("-----BEGIN PUBLIC KEY-----", "")
pem_public_key = pem_public_key.replace("-----END PUBLIC KEY-----", "")
pem_public_key = re.sub("(.{64})", "\\1\n", pem_public_key, 0, re.DOTALL)
pem_public_key = "-----BEGIN PUBLIC KEY-----\n" + pem_public_key
pem_public_key += "\n-----END PUBLIC KEY-----"
key = load_pem_public_key(bytes(pem_public_key), backend=default_backend())
```

This works all fine; as there are no errors coming up. Then, moving on to the encryption part:

```py
if isinstance(key, rsa.RSAPublicKey):
                ciphertext = key.encrypt(bytes(token),
                                         padding.OAEP(
                                             mgf=padding.MGF1(algorithm=hashes.SHA1()),
                                             algorithm=hashes.SHA1(),
                                             label=None
                                         )
                                         )
```

It fails with an error I do not really understand; or better to speak, what went wrong or how to fix it.

### Traceback:

```pytb
[Tue May 31 11:28:06.433256 2016] [wsgi:error] [pid 35398] Traceback (most recent call last):
[Tue May 31 11:28:06.433264 2016] [wsgi:error] [pid 35398]   File "/home/user2/envs/project1/lib/python2.7/site-packages/django/core/handlers/base.py", line 132, in get_response
[Tue May 31 11:28:06.433269 2016] [wsgi:error] [pid 35398]     response = wrapped_callback(request, *callback_args, **callback_kwargs)
[Tue May 31 11:28:06.433272 2016] [wsgi:error] [pid 35398]   File "/home/user2/envs/project1/lib/python2.7/site-packages/django/views/decorators/csrf.py", line 58, in wrapped_view
[Tue May 31 11:28:06.433276 2016] [wsgi:error] [pid 35398]     return view_func(*args, **kwargs)
[Tue May 31 11:28:06.433280 2016] [wsgi:error] [pid 35398]   File "/home/user2/envs/project1/lib/python2.7/site-packages/django/views/generic/base.py", line 71, in view
[Tue May 31 11:28:06.433285 2016] [wsgi:error] [pid 35398]     return self.dispatch(request, *args, **kwargs)
[Tue May 31 11:28:06.433288 2016] [wsgi:error] [pid 35398]   File "/home/user2/envs/project1/lib/python2.7/site-packages/rest_framework/views.py", line 466, in dispatch
[Tue May 31 11:28:06.433292 2016] [wsgi:error] [pid 35398]     response = self.handle_exception(exc)
[Tue May 31 11:28:06.433295 2016] [wsgi:error] [pid 35398]   File "/home/user2/envs/project1/lib/python2.7/site-packages/rest_framework/views.py", line 463, in dispatch
[Tue May 31 11:28:06.433298 2016] [wsgi:error] [pid 35398]     response = handler(request, *args, **kwargs)
[Tue May 31 11:28:06.433330 2016] [wsgi:error] [pid 35398]   File "/var/www/vhosts/secure_api/api/api/hosts/../../account/views.py", line 666, in post
[Tue May 31 11:28:06.433335 2016] [wsgi:error] [pid 35398]     encrypted_token, status_enc = self.encrypt_token(token.private_token, customer)
[Tue May 31 11:28:06.433339 2016] [wsgi:error] [pid 35398]   File "/var/www/vhosts/secure_api/api/api/hosts/../../account/views.py", line 702, in encrypt_token
[Tue May 31 11:28:06.433342 2016] [wsgi:error] [pid 35398]     label=None
[Tue May 31 11:28:06.433345 2016] [wsgi:error] [pid 35398]   File "/home/user2/envs/project1/lib/python2.7/site-packages/cryptography/hazmat/backends/openssl/rsa.py", line 591, in encrypt
[Tue May 31 11:28:06.433348 2016] [wsgi:error] [pid 35398]     return _enc_dec_rsa(self._backend, self, plaintext, padding)
[Tue May 31 11:28:06.433351 2016] [wsgi:error] [pid 35398]   File "/home/user2/envs/project1/lib/python2.7/site-packages/cryptography/hazmat/backends/openssl/rsa.py", line 76, in _enc_dec_rsa
[Tue May 31 11:28:06.433354 2016] [wsgi:error] [pid 35398]     return _enc_dec_rsa_pkey_ctx(backend, key, data, padding_enum)
[Tue May 31 11:28:06.433357 2016] [wsgi:error] [pid 35398]   File "/home/user2/envs/project1/lib/python2.7/site-packages/cryptography/hazmat/backends/openssl/rsa.py", line 105, in _enc_dec_rsa_pkey_ctx
[Tue May 31 11:28:06.433360 2016] [wsgi:error] [pid 35398]     _handle_rsa_enc_dec_error(backend, key)
[Tue May 31 11:28:06.433363 2016] [wsgi:error] [pid 35398]   File "/home/user2/envs/project1/lib/python2.7/site-packages/cryptography/hazmat/backends/openssl/rsa.py", line 128, in _handle_rsa_enc_dec_error
[Tue May 31 11:28:06.433366 2016] [wsgi:error] [pid 35398]     assert errors
[Tue May 31 11:28:06.433369 2016] [wsgi:error] [pid 35398] AssertionError
```

**Comment by tulpn** [comment]
I solved my issue. Maybe it is obvious to you guys, but I leave it here nontheless for people like me working with Django/REST. 

I wrote Testcode:
`pkk = u'-----BEGIN PUBLIC KEY-----MIICIjANBgkqhkiG9w0BAQEFAAOCAg8AMIICCgKCAgEAy2SBwad@TZRyCPDUdxZkGqBipyY7UmTTdGOKL/dgOOsuITXFy/COCi2ylo/owI7u@/K5lkdoBunn1OEu@UFzxRy4bKQK2OkpP6dtAHfyexracpeiHVccRIkQB743DpfupDVtR5TweZsjkRM2SDcpwXZJbnSpa8pzxl6mHvZhV/D2kDkSPLQnI6iWchExt5e/psorpWvy22smaL1xvINztaaRSXFqMQILtSxv/2lJJGc6PRyQacQ4XxDk2DwKBKPyFYHYelYpjGFErUIRz7/EdY4vc3pzr2HMo0AFvvu51ezKIrFGDXt9I7xwEEOSY59mNtA9u9oPyd7UhXkK1JB7OSI458rfHZEXUsvmqqpHH8FKC4dBCZasEHqzMdter05h/v2hMbPV/QgqrArs1p/P6KMpnlYvhnTrIj9p30kOIq7WSfCyXBlTjAb8unYST6vtHXB21TOM6e6szea8fWgCnO3e/xD4M/xdVEqgGxDJWr7qOeHL@l1I93f0wBpO4qlmdMgFz@UqO2p@7dtN4JQO5PwCYHOLgE9bJJZKzaB61BeM4xznqiVfUxOBmtN0k5m1DnKL0lUagSyitwq0H7AY802O91n7ykFYO/gsFkfqNFtdgsGX56lqILU1dYKX3fVsd8r5rENndBE7J0HEUjYz9JHjS@m8EMJ@57kKztP0EgUCAwEAAQ==-----END PUBLIC KEY-----'
pem_public_key = str(pkk).replace("@", "+").strip()
pem_public_key = pem_public_key.replace("-----BEGIN PUBLIC KEY-----", "")
pem_public_key = pem_public_key.replace("-----END PUBLIC KEY-----", "")
pem_public_key = re.sub("(.{64})", "\\1\n", pem_public_key, 0, re.DOTALL)
pem_public_key = "-----BEGIN PUBLIC KEY-----\n" + pem_public_key
pem_public_key += "\n-----END PUBLIC KEY-----\n"
print '[#] Load Public Key from File'

pkey = load_pem_public_key(pem_public_key, backend=default_backend())
if isinstance(pkey, rsa.RSAPublicKey):
	print 'Key Loaded success.....'


print '[#] Encrypting Message'
message = "This is my plaintext message!!!"
print message

ciphertext = pkey.encrypt(
    message,
    padding.OAEP(
        mgf=padding.MGF1(algorithm=hashes.SHA1()),
        algorithm=hashes.SHA1(),
        label=None
    )
)

print '[#] Cipher Text: '
print ciphertext.encode('hex')`

In a virtualenv with the latest version of this library installed. It is pretty much the same code as I was using in my project. Suprisingly everything worked as expected. However, the real code still didn't work. So I made a test method in my real code, commented everything and ran my test func with this exact above code. Still not working.

So, from previous experience working in C# with crypto I figured I should run the code in another thread. 

`thread = Thread(target=self.testing,)
 thread.start()
 thread.join()`

Made everything work. The entire code got executed and the correct response is sent. 

It was a long way from the "Assert failed" error message.
